### PR TITLE
Specify Ruby version

### PR DESCRIPTION
**Pull request by domoscargin**
For Netlify builds, some of our dependencies are [complaining about the Ruby version](https://github.com/alphagov/govuk-frontend-docs/pull/267):

`activesupport-7.0.4 requires ruby version >= 2.7.0, which is incompatible with the current version, ruby 2.6.2p47`

Which is fair, since 2.6.2 is EOL!

## Specifying a Ruby version
Netlify allows specification of the ruby version either through setting an environment variable, or by adding a `.ruby-version` file. This PR implements the latter.

I've specified the latest "normal maintenance" version of Ruby: 3.1.3.

## Rubocop offenses
As a result of bumping Ruby, I've had to update `rubocop-govuk` - the version we were using was only compatible with Ruby up to 2.7, which has entered "security maintenance".

This in turn has thrown up a bunch of rubocop offenses. I've run a safe autocorrect to address most of these. For the rest, I've currently disabled the rule which was causing trouble, but would appreciate somebody with Ruby knowledge to take a look and see if we can quickly fix those. I'll leave comments on the relevant lines.

### CODE REVIEW

File: spec/sassdocs_helpers_spec.rb [hunk 1]
```diff
@@ -13,10 +13,13 @@ def dothash(hash)
   before(:each) do
     allow(File).to receive(:read).and_return('{ "dependencies": { "govuk-frontend": { "version": "1.0.0" } } }')
     # Include mixin into a test class to allow us to mock File
+    # TODO Move constant definition
+    # rubocop:disable Lint/ConstantDefinitionInBlock
```
Comment (domoscargin) [inline spec/sassdocs_helpers_spec.rb:17]: Can this constant definition be edited?
https://msp-greg.github.io/rubocop/RuboCop/Cop/Lint/ConstantDefinitionInBlock.html

File: Rakefile [hunk 2]
```diff
@@ -2,6 +2,7 @@ require "rspec/core/rake_task"
 
 RSpec::Core::RakeTask.new(:spec)
 
+desc "Lint ruby and JavaScript files"
```
Comment (colinrotherham) [inline Rakefile:5]: Probably should title case Ruby like JavaScript

```suggestion
desc "Lint Ruby and JavaScript files"
```

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -44,57 +44,55 @@
...
-  #     --seed 1234
-  config.order = :random
-
-  # Seed global randomization in this process using the `--seed` CLI option.
-  # Setting this allows you to use `--seed` to deterministically reproduce
-  # test failures related to randomization by passing the same `--seed` value
-  # as the one that triggered the failure.
-  Kernel.srand config.seed
-=end
+  # The settings below are suggested to provide a good initial experience
```
Comment (colinrotherham) [inline spec/spec_helper.rb:47]: @domoscargin I feel like we should ignore changes to this file give then message at the top?

```rb
# This file was generated by the `rspec --init` command. Conventionally, all
# specs live under a `spec` directory, which RSpec adds to the `$LOAD_PATH`.
# The generated `.rspec` file contains `--require spec_helper` which will cause
# this file to always be loaded, without a need to explicitly require it in any
# files.
```

Comment (colinrotherham) [reply spec/spec_helper.rb:47]: Looks like [`rspec`](https://github.com/rspec/rspec-metagem#readme) has been replaced with [`rspec-rails`](https://github.com/rspec/rspec-rails#readme)

So yeah probably leave this file alone (ignore comment?) and plan for an update one day

># RSpec
>
>Behaviour Driven Development for Ruby
>
>**The rspec metagem repository has been renamed to rspec-metagem, please update
any rspec/rspec Github references to rspec/rspec-metagem, this is in preparation
for a new mono-repo approach to RSpec dev to unify issue tracking and PR management**

**Review by colinrotherham — APPROVED** [review]
Approved with some tiny comments

File: lib/sassdocs_helpers.rb [hunk 4]
```diff
@@ -41,6 +41,8 @@ def inline_parameters(parameters)
     # Generate the possible parameters that could be passed into a function or mixin.
     # For example with the `govuk-colour` function: `$colour, $legacy`
     # If the function / mixin has default values output them too, for example `$important: true`
+    # TODO Replace concatenation with interpolation
+    # rubocop:disable Style/StringConcatenation
```
[the remark this answers is not in the document]
Comment (domoscargin) [reply lib/sassdocs_helpers.rb:45]: Going to leave these as disabled for now.

File: spec/sassdocs_helpers_spec.rb (hunk 1 above)
Comment (domoscargin) [reply spec/sassdocs_helpers_spec.rb:17]: Going to leave these as disabled for now.

Comment (colinrotherham) [reply spec/sassdocs_helpers_spec.rb:17]: I was happy with these TODO comments by the way 👍

File: spec/spec_helper.rb (hunk 3 above)
Comment (domoscargin) [reply spec/spec_helper.rb]: Donezos

**Pull request merged**